### DISCUSSION TITLE
PUSH WebRTC streams to other instances of MTX.

### DISCUSSION THREAD

**Discussion by benbullnz** [bluenviron/mediamtx#2879] [General]
Starting a discussion about a potential feature which would allow the user to PUSH exisitng webRTC streams in MTX to another instance of MTX/WebRTC server which supports whip.

Currently I have two instances of MTX. Instance 1 hosts a WebRTC stream. Instance 2 can pull that stream using whep/whip. 
It would be great if we could PUSH the stream from instance 1 using WHEP to instance 2.
 
Another option is that there is a utility to push to the 2nd instance which is run with runOnReady?

**Comment by aler9** [comment]
Hello,

Regarding pushing, we're currently not interested into implementing it as a native feature, since FFmpeg or GStreamer inside `runOnReady` are already enough for performing the task.

Regarding an external utility, you can use a recent version of GStreamer and [whipsink](https://gstreamer.freedesktop.org/documentation/webrtchttp/whipsink.html?gi-language=c).

**Comment by rsnk96** [comment]
Hi @aler9 

Does this mean mediamtx cannot act as a WHIP client publishing to a WHIP server hosted elsewhere? (I presume acting as a WHIP client is different from WHEP, as OP asked)

**Reply by aler9** [reply]
@rsnk96 you can use GStreamer and whipclientsink, as explained in the README:
https://github.com/bluenviron/mediamtx?tab=readme-ov-file#gstreamer

**Comment by aler9** [comment]
moved in #4273
